packages/activity/src/index.ts

```diff
@@ -15,23 +15,37 @@
  *
  * ### Cancellation
  *
- * Activity Cancellation serves three purposes:
+ * Activity Cancellation:
  *
- * - It lets an Activity know it doesn't need to keep doing work.
- * - It gives the Activity time to clean up any resources it has created.
+ * - lets the Activity know it doesn't need to keep doing work, and
+ * - gives the Activity time to clean up any resources it has created.
  *
- * Activities may receive Cancellation only if they {@link Context.heartbeat | emit heartbeats} or are Local Activities
+ * Activities can only receive Cancellation if they {@link Context.heartbeat | emit heartbeats} or are Local Activities
  * (which can't heartbeat but receive Cancellation anyway).
  *
- * There are two ways to handle Activity cancellation:
+ * An Activity may receive Cancellation if:
+ *
+ * - The Workflow was Cancelled (via {@link WorkflowHandle.cancel}) and {@link ActivityOptions.cancellationType} was
+ *   **not** set to {@link ActivityCancellationType.ABANDON}.
+ * - The Activity was manually Cancelled (it was called inside a {@link CancellationScope} that was Cancelled via
+ *   {@link CancellationScope.cancel}).
+ * - The Worker has started to shut down. Shutdown is initiated by either:
+ *   - One of the {@link RuntimeOptions.shutdownSignals} was sent to the process.
+ *   - {@link Worker.shutdown | `Worker.shutdown()`} was called.
+ * - The Activity was considered failed by the Server because the Server didn't receive a heartbeat within the
+ *   {@link ActivityOptions.heartbeatTimeout}. The {@link CancellationFailure} will have `message: 'TIMED_OUT'`.
+ * - The Workflow was terminated, in which case the {@link CancellationFailure} will have `message: 'NOT_FOUND'`.
```

Comment:
This can be generalized to say the workflow *run* has reached a "final" state completed, terminated, failed, and so forth.

Reply:
We might also cancel an activity if it sends heartbeat details that cannot be converter by the worker's data converter.

Reply:
I'd also mention that the cancel reason is provided in the CancelledFailure thrown in the activity and is accessible via the abort signal (don't remember exactly how to do the latter).